awx/api/views.py

```diff
@@ -2941,14 +2938,29 @@ def get_queryset(self):
 
     def is_valid_relation(self, parent, sub, created=False):
         current_extra_types = [
-            cred.credential_type.pk for cred in parent.extra_credentials.all()
+            cred.credential_type.pk for cred in parent.credentials.all()
+            if cred.kind not in 'vault'
         ]
         if sub.credential_type.pk in current_extra_types:
             return {'error': _('Cannot assign multiple %s credentials.' % sub.credential_type.name)}
 
-        if sub.credential_type.kind not in ('net', 'cloud'):
+        return super(JobTemplateCredentialsList, self).is_valid_relation(parent, sub, created)
+
+
+class JobTemplateExtraCredentialsList(JobTemplateCredentialsList):
+
+    deprecated = True
+
+    def get_queryset(self):
+        sublist_qs = super(JobTemplateExtraCredentialsList, self).get_queryset()
```

Comment:
The `extra_credentials` endpoint is now just a subset of the `credentials` endpoint - it works exactly the same way, expect it filters on `kind in ('cloud', 'net')`.